### PR TITLE
Fix zio-http authentication example

### PR DESCRIPTION
**Pull request by guizmaii**

### CODE REVIEW

File: examples/src/main/scala/example/ziohttp/AuthExampleApp.scala [hunk 1]
```diff
@@ -52,7 +61,7 @@ object Auth {
               case _                              => None
             })
             .orElseFail(CalibanError.ExecutionError("Unable to decode payload"))
-            .flatMap(user => ZIO.service[Auth].flatMap(_.setUser(user).orDie))
+            .flatMap(auth.setUser)
```
Comment (guizmaii) [inline examples/src/main/scala/example/ziohttp/AuthExampleApp.scala:64]: It was not the correct `Auth` instance used

File: examples/src/main/scala/example/ziohttp/AuthExampleApp.scala [hunk 2]
```diff
@@ -7,39 +7,48 @@ import caliban.interop.tapir.{ StreamTransformer, WebSocketHooks }
...
+          def setUser(name: String): UIO[Unit]      = ref.set(name)
+        }
       }
-    }
-    .toLayer
+      .toLayer
 
   object WebSockets {
-    val wsSession = Http.fromEffect(Ref.make[String]("unknown"))
+    private val wsSession = Http.fromEffect(Ref.make[String](null))
```
Comment (ghostdogpr) [inline examples/src/main/scala/example/ziohttp/AuthExampleApp.scala:42]: How about `Option[String]` and no null 😄

Comment (guizmaii) [reply examples/src/main/scala/example/ziohttp/AuthExampleApp.scala:42]: `Option` creates a useless allocation. In terms of performances, `null` is better solution here

Comment (ghostdogpr) [reply examples/src/main/scala/example/ziohttp/AuthExampleApp.scala:42]: It's insignificant compared to everything that happens when handling a query. I think it's better that examples look cleaner.

Comment (guizmaii) [reply examples/src/main/scala/example/ziohttp/AuthExampleApp.scala]: Changed for Option

**Review by frekw — APPROVED** [review]
Nice!

**Comment by ghostdogpr** [comment]
Thanks!

**Pull request merged**